lib/datasource/terraform-provider/index.ts

```diff
@@ -51,23 +73,54 @@ export async function getReleases({
     dep.releases = res.versions.map((version) => ({
       version,
     }));
-    if (pkgUrl.startsWith('https://registry.terraform.io/')) {
-      dep.homepage = `https://registry.terraform.io/providers/${repository}`;
-    }
+    dep.homepage = `https://registry.terraform.io/providers/${repository}`;
     logger.trace({ dep }, 'dep');
     const cacheMinutes = 30;
     await globalCache.set(cacheNamespace, pkgUrl, dep, cacheMinutes);
     return dep;
   } catch (err) {
-    if (err.statusCode === 404 || err.code === 'ENOTFOUND') {
-      logger.debug(
-        { lookupName },
-        `Terraform registry lookup failure: not found`
-      );
-      logger.debug({
-        err,
-      });
-      return null;
+    if (
+      err.statusCode === 404 ||
+      err.code === 'ENOTFOUND' ||
+      err.stack.startsWith('TypeError')
+    ) {
+      try {
+        // registry returns an error, falling back to releases.hashicorp.com
+        const res = (
```

Comment:
I suggest it's best to move this "catch" logic into a separate function that's called from here.